core/block.js

```diff
@@ -1445,3 +1445,19 @@ Blockly.Block.prototype.allInputsFilled = function(opt_shadowBlocksAreFilled) {
 
   return true;
 };
+
+/**
+ * @return {string} A description of this block using type name and id.
+ */
+Blockly.Block.prototype.describe = function() {
+  var msg;
+  if (this.type) {
+    msg = '"' + this.type + '" block';
+  } else {
+    msg = 'Block';
+  }
```

Comment:
Consider these six lines might be easier to read if inlined:
`var msg = this.type ? '"' + this.type + '" block' : 'Block';`